server/src/testFixtures/java/io/crate/testing/SQLTransportExecutor.java

```
@@ -499,48 +497,80 @@ private BulkResponse executeBulk(String stmt, Object[][] bulkArgs, TimeValue tim
-            // execute before the second cluster state update removes the old master and the condition ensureGreen / ensureYellow will
-            // trivially hold if it held before the node was shut down. The following "waitForNodes" condition ensures that the node has
-            // been removed by the master so that the health check applies to the set of nodes we expect to be part of the cluster.
-            request.waitForNodes(Integer.toString(expectedNumNodes));
-        }
-        ClusterHealthResponse actionGet = FutureUtils.get(client.admin().cluster().health(request));
-        if (state == ClusterHealthStatus.YELLOW) {
-            assertThat(actionGet.getStatus()).satisfiesAnyOf(
-                s -> assertThat(s).isEqualTo(state),
-                s -> assertThat(s).isEqualTo(ClusterHealthStatus.GREEN));
-        } else {
-            assertThat(actionGet.getStatus()).isEqualTo(state);
-        }
-        return actionGet.getStatus();
+    @SuppressWarnings("unchecked")
+    private void ensureState(ClusterHealthStatus state, @Nullable Integer expectedNumNodes) throws Exception {
+        ESTestCase.assertBusy(() -> {
+            // First, check the global cluster health. If e.g. the cluster hasn't elected a master node, the 2nd query
+            // containing the `sys.allocations` table will fail this table is routed to the master node.
+            var clusterHealthResponse = exec("select health, description from sys.cluster_health");
+            String color = (String) clusterHealthResponse.rows()[0][0];
+            String description = (String) clusterHealthResponse.rows()[0][1];
+            var assertion = assertThat(color)
+                .withFailMessage("Expected cluster to be %s, but was %s, reason: %s", state.name(), color, description);
+            if (state == ClusterHealthStatus.GREEN) {
+                assertion.isEqualTo(state.name());
+            } else {
```

minor: I see that currently we don't have a call with `RED`, so to make the code more robust, maybe throw an error before even the `assertBusy()` that `RED` is illegal?

---

will add an assertion.